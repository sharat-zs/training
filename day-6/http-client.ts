import axios from 'axios'

export const URL = 'https://jsonplaceholder.typicode.com/todos/1'
export async function fetchData(url: string) {
    const res = await axios.get(url)
    const data = res.data
    const status = res.status
    return { data, status }
}

// fetchData(URL).then((res) => console.log(res))
