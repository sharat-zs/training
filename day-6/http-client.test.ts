import axios from 'axios'
import AxiosMockAdapter from 'axios-mock-adapter'
import { fetchData, URL } from './http-client'

// jest.mock('axios')
// const mockedAxios = axios as jest.Mocked<typeof axios>
const mockedAxios = new AxiosMockAdapter(axios)
mockedAxios.onGet(URL).reply(200, {
    users: [{ id: 1, name: 'John Smith' }]
})

describe('http-client', () => {
    it('should return proper response', async () => {
        const { status, data } = await fetchData(URL)
        expect(status).toBe(200)
        expect(data).toMatchSnapshot()
    })
})
