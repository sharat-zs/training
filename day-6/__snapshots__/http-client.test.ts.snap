// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http-client should return proper response 1`] = `
Object {
  "users": Array [
    Object {
      "id": 1,
      "name": "John Smith",
    },
  ],
}
`;
